### PR TITLE
Fixes warp facing direction. Issue #1240

### PR DESCRIPTION
**Pull request by KirieZ**
This fixes issue #1240 by changing facing direction when setting character position.

---
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1353)

### CODE REVIEW

**Comment by hemagx** [comment]
While it's probably bugged aegis behavior would not be better to update
direction on player side to fix the issue than force north facing?

**Comment by MishimaHaruna** [comment]
Or it could be made configurable (via battle_config flag)

**Comment by KirieZ** [comment]
I'll take a look, I thought it was forced by the client and as other commits were fixing it that way I just followed.

**Comment by KirieZ** [comment]
I like the idea of making it configurable, as we add aegis behavior and at the same time one that make more sense. It's possible to change the direction after warping, but to keep it after logging out we need a new column, what should I do? where is the best place to put this on battle conf? player.conf? and what should I call it? Thanks

**Comment by hemagx** [comment]
It's useless to log this, keep it reset on relog but only remembered when
warping

**Comment by KirieZ** [comment]
Ok, done.

**Comment by kyeme** [comment]
Why not set to default?

**Comment by Emistry** [comment]
instead of keep the facing direction or not, why not just use the config to set the default direction after warped?

enable it to set for all 8 directions.
set the facing direction according to the config value.

File: conf/battle/player.conf [hunk 1]
```diff
@@ -185,3 +185,7 @@ idletime_criteria: 0x1F
 costume_refine_def: yes
 shadow_refine_def: yes
 shadow_refine_atk: yes
+
+// Keep player facing direction after warping?
+// Default: no (on official servers players always faces north)
+player_warp_keep_direction: yes
```
Comment (hemagx) [inline conf/battle/player.conf]: Please add a new line

**Comment by hemagx** [comment]
@kyeme cause no sense Aegis behavior doesn't deserve to bet set as default :)
@Emistry It's another customize suggestion, but it's no sense to customize facing at all.

File: src/map/clif.c [hunk 2]
```diff
@@ -9547,7 +9547,8 @@ void clif_parse_LoadEndAck(int fd, struct map_session_data *sd) {
 		clif->clearunit_area(&sd->bl, CLR_DEAD);
 	else {
 		skill->usave_trigger(sd);
-		sd->ud.dir = 0;/* enforce north-facing (not visually, virtually) */
+		if (battle_config.player_warp_keep_direction)
```
Comment (hemagx) [inline src/map/clif.c]: kindly use == 1 here please

File: src/map/pc.c [hunk 3]
```diff
@@ -5580,6 +5580,8 @@ int pc_setpos(struct map_session_data* sd, unsigned short map_index, int x, int
 		npc->script_event(sd, NPCE_LOGOUT);
 		//remove from map, THEN change x/y coordinates
 		unit->remove_map_pc(sd,clrtype);
+		if (!battle_config.player_warp_keep_direction)
+			sd->ud.dir = 0; // makes character face north
```
Comment (hemagx) [inline src/map/pc.c]: same here, please use == 0

File: src/map/pc.c [hunk 4]
```diff
@@ -5610,6 +5612,9 @@ int pc_setpos(struct map_session_data* sd, unsigned short map_index, int x, int
 		vending->close(sd);
 	}
 
+	if (!battle_config.player_warp_keep_direction)
```
Comment (hemagx) [inline src/map/pc.c]: same as above

**Comment by KirieZ** [comment]
Maybe it would be better to not use "Default" word on the comment? Just say that official servers does that? I think what @kyeme is saying is that we should be using the comment's default value. I think I didn't use it because I tested with yes and forgot to change back to no before commiting.

**Pull request closed without merging**